### PR TITLE
Dialogflow - Knowledge

### PR DESCRIPTION
**Pull request by nirupa-kumar**
@nnegrey Please review

DetectIntentKnowledge
KnowledgeManagement
DocumentManagement

### CODE REVIEW

File: samples/src/main/java/com/example/dialogflow/DetectIntentKnowledge.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import java.util.UUID;
+
+// [END dialogflow_import_libraries]
+
+/** DialogFlow API Detect Intent sample with querying knowledge connector. */
+public class DetectIntentKnowledge {
+
+  // [START dialogflow_detect_intent_knowledge]
+  /**
+   * Returns the result of detect intent with streaming audio as input.
```
Comment (nnegrey) [inline samples/src/main/java/com/example/dialogflow/DetectIntentKnowledge.java]: Update the comment

File: samples/src/main/java/com/example/dialogflow/DetectIntentKnowledge.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+   *
+   * <p>Using the same `session_id` between requests allows continuation of the conversation.
+   *
+   * @param projectId Project/Agent Id.
+   * @param texts The texts to be processed.
+   * @param sessionId Identifier of the DetectIntent session.
+   * @param languageCode Language code of the query.
+   * @throws Exception
```
Comment (nnegrey) [inline samples/src/main/java/com/example/dialogflow/DetectIntentKnowledge.java]: Add param for knowledgeBaseId

File: samples/src/main/java/com/example/dialogflow/DetectIntentKnowledge.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        // Build the query with the TextInput
+        QueryInput queryInput = QueryInput.newBuilder().setText(textInput).build();
+
+        KnowledgeBaseName knowledgeBaseName = KnowledgeBaseName.of(projectId, knowledgeBaseId);
+        QueryParameters queryParameters =
+            QueryParameters.newBuilder()
+                .addKnowledgeBaseNames(knowledgeBaseName.toString())
+                .build();
+
+        DetectIntentRequest dr =
```
Comment (nnegrey) [inline samples/src/main/java/com/example/dialogflow/DetectIntentKnowledge.java]: Can we switch `dr` --> `detectIntentRequest`?

File: samples/src/main/java/com/example/dialogflow/KnowledgebaseManagement.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  /**
+   * List Knowledge bases
+   *
+   * @param projectId Project/agent id.
+   */
+  public static void listKnowledgeBases(String projectId) throws Exception {
+    // Instantiates a client
+    try (KnowledgeBasesClient knowledgeBasesClient = KnowledgeBasesClient.create()) {
+      // Set the entity type name using the projectID (my-project-id) and entityTypeId (KIND_LIST)
+      ProjectName parent = ProjectName.of(projectId);
```
Comment (nnegrey) [inline samples/src/main/java/com/example/dialogflow/KnowledgebaseManagement.java]: `parent` --> `projectName`

File: samples/src/main/java/com/example/dialogflow/KnowledgebaseManagement.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+   *
+   * @param projectId Project/agent id.
+   * @param displayName Name of the knowledge base.
+   */
+  public static void createKnowledgeBase(String projectId, String displayName) throws Exception {
+    // Instantiates a client
+    try (KnowledgeBasesClient knowledgeBasesClient = KnowledgeBasesClient.create()) {
+
+      KnowledgeBase knowledgeBase = KnowledgeBase.newBuilder().setDisplayName(displayName).build();
+      ProjectName parent = ProjectName.of(projectId);
```
Comment (nnegrey) [inline samples/src/main/java/com/example/dialogflow/KnowledgebaseManagement.java]: `parent` --> `projectName`

File: samples/src/test/java/com/example/dialogflow/KnowledgeBaseIT.java [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+  }
+
+  @After
+  public void tearDown() {
+    System.setOut(null);
+  }
+
+  @Test
+  public void testKnowledgeBase() throws Exception {
+ // Check the knowledge base does not yet exist
```
Comment (nnegrey) [inline samples/src/test/java/com/example/dialogflow/KnowledgeBaseIT.java]: indentation

**Review by nnegrey — CHANGES_REQUESTED** [review]
A couple NIT things.

**Comment by nirupa-kumar** [comment]
@nnegrey 
Thanks Noah. Fixed the issues. Please let me know if there is anything else.

**Review by nnegrey — APPROVED** [review]
LGTM

**Pull request merged**